src/main/java/com/microsoft/graph/http/BaseRequest.java

```diff
@@ -265,7 +267,7 @@ public HttpMethod getHttpMethod() {
     @Override
     @Nullable
     public List<HeaderOption> getHeaders() {
-        return headersOptions;
+        return Collections.unmodifiableList(headersOptions);
```

Comment:
unless I missed something, we have get (list) and add methods for those collections but we don't have a remove method.
Making the collection unmodifiable might break people whom are modifying the list here.
I think we should grand-father these in knowing that with Kiota all of that will go away anyway. What do you think?

Reply:
I do think you're right in that it would most likely break peoples code, even here in this file these changes caused some breaking changes in the addQueryOption and addFunctionOption functions. We could possibly add remove functions, though that wouldn't prevent the breaking changes, but I don't think the upside is truly worth it. I can go ahead and suppress these warnings for now.

Reply:
please do

Reply:
Just for clarification, we do want to revert the changes on getQuery getFunction as well as on getHeaders correct?

Reply:
correct